communication/state_telemetry.cpp

```diff
@@ -203,6 +213,44 @@ static mav_result_t state_telemetry_set_mode_from_cmd(State* state, mavlink_comm
     return result;
 }
 
+static mav_result_t state_telemetry_set_arm_from_cmd(State* state, mavlink_command_long_t* packet)
+{
+    mav_result_t result = MAV_RESULT_ACCEPTED;
+
+    mav_mode_t new_mode;
+    float arm_cmd = packet->param1;
+
+    if (state->mav_state == MAV_STATE_CALIBRATING)
+    {
+        result = MAV_RESULT_TEMPORARILY_REJECTED;
+        print_util_dbg_print("Currently calibrating, impossible to change mode.\r\n");
+    }
+    else
+    {
```

Comment:
I am also in favor of option 2. 
Currently, it is already programmed in the option 2. 

The MAVLink message change the state->mav_mode armed flag. However, when the remote is active, in the [state_machine update function](https://github.com/lis-epfl/MAVRIC_Library/blob/master/communication/state_machine.cpp#L105), we overwrite the mav_mode with the value of the remote armed flag. Therefore, the robot may be switched on for a few loops but is switched off again quite soon. 

That being said, I think, we should implement option 2 in all set_mode message function to be more safe and clean (and not only this callback).